models/image.py

```diff
@@ -905,44 +714,70 @@ def from_ref_and_new(cls, ref, new):
 
         Parameters
         ----------
-        ref: Image object
-            The reference image to use.
-        new: Image object
+        new_image: Image object
             The new image to use.
-
+        ref_image: Image object
+            The reference image to use.
+        im_type: str (optional)
+            If given, will also set the output image's type.
+            Use "diff" or "warped", which will be prepended
+            with "Com" if the new image is a "Com..." type image
+            Note that if not given, the output type will remain as "Sci".
         Returns
         -------
         output: Image
             The new Image object. It would not have any data variables or filepath.
         """
+        if ref_image is None:
+            raise ValueError("Must provide a reference image.")
+        if new_image is None:
+            raise ValueError("Must provide a new image.")
+
         output = Image(nofile=True)
 
         # for each attribute, check the two images have the same value
-        for att in ['section_id', 'instrument', 'telescope', 'filter', 'project', 'target']:
-            ref_value = getattr(ref, att)
-            new_value = getattr(new, att)
+        for att in ['instrument', 'telescope', 'project', 'section_id', 'filter', 'target']:
```

Comment:
See comment above in from_images -- I wouldn't require project, section_id, and target to be the same.  (Requiring that, for instance, would mean we couldn't do what I did with the lensgrinder pipeline, where the references came from a deep survey previuosly performed with DECam.)

Reply:
Unresolved this -- I still think this should be only checking instrument, telescope, and filter.  This is the highest-level, most general thing that associates images.  Since we know there are cases where projects, sections, and targets will be mixed, we shouldn't be putting checks on that here.  If there are places in the more specific implementation functions where those checks are appropriate, put them there, but putting them here limits us a priori in ways we know we don't want to be limited.

Reply:
Will fix this in issue #179